cpu/pic32/lib/pic32_uart.c

```diff
@@ -163,7 +163,15 @@
     U##XX##MODESET = _U##XX##MODE_UARTEN_MASK;                                                                  \
                                                                                                                 \
     return UART_NO_ERROR;                                                                                       \
-  }
+  }                                                                                                             \
+int8_t                                                                                                          \
+pic32_uart##XX##_release()                                                                                      \
+{                                                                                                               \
+  pic32_uart##XX##_power_down();                                                                                \
```

Comment:
Power down needs to be done after register changes, otherwise any register changes will be discarded.